person.rb

```diff
@@ -1,21 +1,27 @@
 require_relative './nameable'
+require_relative './rental'
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(id, age, name = 'Unknown', parent_permission: true)
     super()
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def add_rental(rental)
+    @rentals << rental
+  end
```

Comment:
- [ ] When adding rental here, remember there is a `many-to-many` relationship between the `Person` and the `Book` through the `Rental`, what we need to do is to make sure that the `Rental` constructor has been modified appropriately so that when we instantiate the `Rental` class here in this method, the rental will be added to the `@rentals` array in both `Book` and `Person`. 
**HINT**
```ruby
def add_rental(book, date)
  Rental.new(date, book, self)
end
```